src/java.base/unix/classes/sun/nio/fs/UnixChannelFactory.java

```diff
@@ -195,7 +195,7 @@ protected static FileDescriptor open(int dfd,
 
             // create flags
             if (flags.createNew) {
-                byte[] pathForSysCall = path.asByteArray();
+                byte[] pathForSysCall = path.getByteArrayForSysCalls();
```

Comment:
Does this need to be done at line 240 as well?
```
fd = openat(dfd, path.asByteArray(), oflags, mode);
```

Reply:
When *at functions the second parameter is relative to the open directory so need to be careful, need to make sure there are good tests before changing anything.

Reply:
> When *at functions the second parameter is relative to the open directory [...]

I tested this for `fstatat` by obtaining the attributes of a file via a `SecureDirectoryStream` and it made no difference, but if `UnixFileSystem.needToResolveAgainstDefaultDirectory()` returned `true`, it seems likely it would introduce a problem. Hence I suggest ignoring my comment above and leaving line 240 as is.